ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java

```diff
@@ -2463,6 +2473,15 @@ private RelNode applyPostJoinOrderingTransform(RelNode basePlan, RelMetadataProv
                 HiveProjectSortExchangeTransposeRule.INSTANCE, HiveProjectMergeRule.INSTANCE);
       }
 
+      // 10. We need to expand IN/BETWEEN expressions when loading a materialized view
```

Comment:
Why do we need to call these rules again in post ordering?

Reply:
This is call only when we are loading the materialized view in the registry so we can keep the plan in the correct form (IN and BETWEEN expanded) and we do not have to do it at compilation time.